CMakeLists.txt

```diff
@@ -189,8 +200,6 @@ foreach(prec ${precisions})
       TARGET ${LIBNAME}_${prec}
       SOURCES ${prec_srcs} $<TARGET_OBJECTS:${LIBNAME}>
       DEFINITIONS $<$<NOT:${fiat_FOUND}>:${FIELD_API_DEFINITIONS}>
-      PUBLIC_LIBS
```

Comment:
As discussed offline, I think the fundamental problem is that we want to propagate link flags, but not compiler flags. I tried a suggestion by @reuterbal to switch this to
```
INTERFACE_LIBS
    $<${HAVE_ACC}:OpenACC::OpenACC_Fortran>
```
I think this works for our use case, so potentially the additional configuration option is not needed after all?